link.go

```diff
@@ -111,11 +116,40 @@ func (link *ToxicLink) UpdateToxic(toxic *toxics.ToxicWrapper) {
 func (link *ToxicLink) RemoveToxic(toxic *toxics.ToxicWrapper) {
 	i := toxic.Index
 
-	// Interrupt the last toxic so that the target's buffer is empty
-	if link.stubs[i-1].InterruptToxic() && link.stubs[i].InterruptToxic() {
+	if link.stubs[i].InterruptToxic() {
+		stop := make(chan bool)
+		// Interrupt the previous toxic to update its output
+		go func() {
+			stop <- link.stubs[i-1].InterruptToxic()
+		}()
+
+		// Unblock the previous toxic if it is trying to flush
+		// If the previous toxic is closed, continue flusing until we reach the end.
+		interrupted := false
+		stopped := false
+		for !interrupted {
+			select {
+			case interrupted = <-stop:
+				stopped = true
+			case tmp := <-link.stubs[i].Input:
+				if tmp == nil {
+					link.stubs[i].Close()
+					if !stopped {
+						<-stop
+					}
+					return
+				}
+				link.stubs[i].Output <- tmp
+			}
+		}
+
 		// Empty the toxic's buffer if necessary
 		for len(link.stubs[i].Input) > 0 {
```

Comment:
This part handles the case where the toxic has a buffer size != 0. The above case is only if the end of stream is reached while interrupting.
There might be a way to consolidate this into 1, I'll have a look later.